sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -245,15 +245,22 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
         "read files of Hive data source directly.")
     }
 
+    val updatedPaths = if (paths.length == 1) {
```

Comment:
I think the most intuitive behavior is to drop the `path` option if `load` is called with path parameters, no matter it's one or more paths.

Reply:
If we are worried about silent result changing, we can fail if there are `path` option and `load` is called with path parameters. The error message should ask users to either remove the `path` options, we put it into the `load` parameters.

Reply:
+1 for your suggestion